src/AuthorizationUtils.php

```diff
@@ -0,0 +1,103 @@
+<?php
+
+/*
+ * This file is part of the Mercure Component project.
+ *
+ * (c) Kévin Dunglas <dunglas@gmail.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace Symfony\Bundle\MercureBundle;
+
+use Symfony\Component\HttpFoundation\Cookie;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\Mercure\Exception\InvalidArgumentException;
+use Symfony\Component\Mercure\Hub;
+use Symfony\Component\Mercure\Jwt\TokenFactoryInterface;
+use Symfony\Component\WebLink\GenericLinkProvider;
+use Symfony\Component\WebLink\Link;
+
+final class AuthorizationUtils
+{
+    private const MERCURE_AUTHORIZATION_COOKIE_NAME = 'mercureAuthorization';
+
+    private $hubs;
+    private $factories;
+
+    /**
+     * @param iterable<string, Hub>                   $hubs
+     * @param iterable<string, TokenFactoryInterface> $factories
+     */
+    public function __construct(iterable $hubs, iterable $factories)
+    {
+        $this->hubs = \is_array($hubs) ? $hubs : iterator_to_array($hubs);
+        $this->factories = \is_array($factories) ? $factories : iterator_to_array($factories);
```

Comment:
`iterator_to_array()` breaks laziness here, which is the main purpose of `IteratorArgument`.

Using a `ServiceLocatorArgument` instead would allow accessing hubs by key while keeping them lazily loaded. 
You can use `ServiceLocatorTagPass::register()` to create one, see e.g. https://github.com/symfony/symfony/blob/5.x/src/Symfony/Component/Validator/DependencyInjection/AddConstraintValidatorsPass.php